### PR TITLE
Ndlano/issues#1388 new subjectpage design

### PR DESCRIPTION
**Pull request by chrpeter**
Resolves #1388

This pull request adds support for layouts instead of display in two columns. It supports three different layouts: `single`, `double` and `stacked`.

Following should work:
- [ ] Supports `layout` instead of `displayInTwoColumns`
- [ ] Single should show everything in one column
- [ ] Double should show everything in two columns
- [ ] Stacked should show everything under filters as headings in separate sections. Each sections should be two columns.

### CODE REVIEW

File: src/containers/SubjectPage/components/SubjectPageContent.jsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  subject: GraphQLSubjectShape,
+};
+
+const SubjectPageContent = ({ layout, subject, ...rest }) => {
+  const breadcrumb = <SubjectBreadCrumb subject={subject} />;
+  const topics =
+    subject && subject.topics
+      ? subject.topics
+          .filter(
+            topic => !topic || !topic.parent || topic.parent === subject.id,
```
Comment (oyvinmar) [inline src/containers/SubjectPage/components/SubjectPageContent.jsx]: Is this nesseary after [this](https://github.com/NDLANO/ndla-frontend/pull/265/files) change?

File: src/containers/SubjectPage/components/SubjectPageContent.jsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+          )
+          .map(topic => toTopicMenu(topic, subject.topics))
```
Comment (oyvinmar) [inline src/containers/SubjectPage/components/SubjectPageContent.jsx]: Maybe drop using toTopicManu and just map topic meta to introduction?

File: src/containers/SubjectPage/components/layout/SubjectPageStacked.jsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+      handleFilterClick,
+      filters,
+      breadcrumb,
+      subjectId,
+      activeFilters,
+      t,
+    } = this.props;
+    const { editorsChoices } = subjectpage;
+    const sortedTopics = this.getSortedTopics();
+    // TODO: FIX TRANSLATION
```
Comment (oyvinmar) [inline src/containers/SubjectPage/components/layout/SubjectPageStacked.jsx]: ?

File: src/containers/SubjectPage/components/layout/SubjectPageStacked.jsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+                options={filters}
+                values={activeFilters}
+                onChange={handleFilterClick}
+              />
+              {sortedTopics.map(filterTopics => (
+                <div {...classes()} key={filterTopics.key}>
+                  <h1 {...classes('heading')}>{filterTopics.heading}</h1>
+                  <TopicIntroductionList
+                    toTopic={toTopic(subjectId, activeFilters)}
+                    topics={filterTopics.topics.sort(
```
Comment (oyvinmar) [inline src/containers/SubjectPage/components/layout/SubjectPageStacked.jsx]: Check if sorting is fixed in taxonomy-api in the spoletest environment before adding sorting here.

File: src/containers/SubjectPage/components/layout/SubjectPageStacked.jsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  prefix: 'c-',
+});
+
+class SubjectPageStacked extends React.PureComponent {
+  constructor() {
+    super();
+    this.getSortedTopics = this.getSortedTopics.bind(this);
+  }
+
+  getSortedTopics() {
```
Comment (oyvinmar) [inline src/containers/SubjectPage/components/layout/SubjectPageStacked.jsx]: Confusing fn name. It does'n do any sorting?

**Comment by chrpeter** [comment]
Fixed now @oyvinmar

**Pull request merged**